server/src/main/java/org/elasticsearch/plugins/internal/DocumentSizeObserver.java

```diff
@@ -40,7 +41,16 @@ public long normalisedBytesParsed() {
 
     /**
      * Returns the state gathered during parsing
+     *
      * @return a number representing a state parsed
      */
     long normalisedBytesParsed();
+
+    default IndexRequest setNormalisedBytesParsedOn(IndexRequest indexRequest) {
```

Comment:
You are ignoring the return value in most places anyway, and there is a strong feeling that this method will have side effects in the implementations.
Can we return void to make this more explicit? And maybe add some javadocs?

Reply:
Strong agree on above 👍 +1 on Javadocs

Reply:
fair point, we no logner use this method as a 'chained fluent builder'. will remove to void
I will add simple javadoc. but the details that are important are in serverless code (linked to this PR)